src/plugins/blockbook.ts

```diff
@@ -0,0 +1,92 @@
+import WebSocket from 'ws'
+import { makeEvents } from 'yavent'
+
+import { messageToString } from '../messageToString'
+import { AddressPlugin, PluginEvents } from '../types/addressPlugin'
+import { blockbookProtocol } from '../types/blockbookProtocol'
+
+export interface BlockbookOptions {
+  pluginId: string
+
+  /** The actual connection URL */
+  url: string
+}
+
+export function makeBlockbook(opts: BlockbookOptions): AddressPlugin {
+  const { pluginId, url } = opts
+
+  const ws = new WebSocket(url)
+  const [on, emit] = makeEvents<PluginEvents>()
+  const codec = blockbookProtocol.makeClientCodec({
+    handleError(error) {
+      emit('error', error)
+    },
+    async handleSend(text) {
+      ws.send(text)
+    },
+    localMethods: {
+      subscribeAddresses({ address }) {
+        emit('update', { address })
+      }
+    }
+  })
+
+  ws.on('message', message => {
+    const text = messageToString(message)
+    codec.handleMessage(text)
+  })
+  ws.on('open', () => {
+    emit('connect', undefined)
+  })
+  ws.on('close', () => {
+    codec.handleClose()
+    emit('disconnect', undefined)
+  })
```

Comment:
This plugin works fine as long as this WebSocket is up & running. As soon as the connection drops, the plugin will die and never work again.

We need to have the plugin start trying to reconnect as soon as this happens, maybe setting `ws = undefined` and `coded = undefined`, and then creating a new WebSocket + codec pair to replace the failed one after some timeout.

Reply:
I've refactored to multiple connections, so this is outdated. However, perhaps reconnect isn't implemented properly there. Let's review that commit and redo feedback.

Additionally, we may pivot from using Blockbook for ETH and prioritize a Etherscan plugin because mempool blockbook txs are problematic for the current accountbased plugin as of today.